src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -148,32 +148,61 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
             ILspLogger logger,
             RequestTelemetryLogger telemetryLogger,
             ILspWorkspaceRegistrationService lspWorkspaceRegistrationService,
+            LspMiscellaneousFilesWorkspace lspMiscFilesWorkspace,
             TextDocumentIdentifier textDocument,
             string? clientName)
         {
             logger.TraceInformation($"Finding document corresponding to {textDocument.Uri}");
 
-            using var workspaceKinds = TemporaryArray<string?>.Empty;
-            foreach (var workspace in lspWorkspaceRegistrationService.GetAllRegistrations())
+            var document = TryGetDocumentFromRegisteredWorkspaces(textDocument, clientName, lspWorkspaceRegistrationService, telemetryLogger, logger);
+            if (document != null)
             {
-                workspaceKinds.Add(workspace.Kind);
-                var documents = workspace.CurrentSolution.GetDocuments(textDocument.Uri, clientName, logger);
+                return document;
+            }
 
-                if (!documents.IsEmpty)
+            // If the document was not in a registered workspace, try to retrieve from the LSP misc files workspace.
+            document = TryGetDocumentFromWorkspace(textDocument, clientName, lspMiscFilesWorkspace);
+            return document;
```

Comment:
Should we still register the misc files workspace? Or is it too special at this point?

Reply:
this regular misc files workspace _is_ registered in VS (so we should have already found the file there).  But only for VS since it doesn't work anywhere else.